tests/components/media_player/test_kef.py

```diff
@@ -0,0 +1,234 @@
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        return self.__source
+
+    @source.setter
+    def source(self, value):
+        """Mock for setting the input source."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        self.__source = value
+
+    @property
+    def muted(self):
+        """Mock for muted. True if muted, else False."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        return self.__muted
+
+    @muted.setter
+    def muted(self, value):
+        """Mock for set muted."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        self.__muted = value
+
+    @property
+    def online(self):
+        """Mock for get if speaker is online or not."""
+        return self.is_online
+
+    # pylint: disable=invalid-name
+    def turnOff(self):
+        """Mock for turn off the speaker."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        self.is_online = False
+        return True
+
+    # pylint: disable=invalid-name
+    def increaseVolume(self, step=None):
+        """Mock for increase volume."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        if not self.__muted:
+            self.__volume = self.__volume + (step if step else 0.05)
+
+    # pylint: disable=invalid-name
+    def decreaseVolume(self, step=None):
+        """Mock for decrease volume."""
+        self.increaseVolume(-(step or 0.05))
+
+
+def add_entities_factory(self):
+    """Add devices factory."""
+    def add_entities(devices, update_befor_add=False):
+        """Fake add device."""
+        self.devices += devices
+
+    return add_entities
+
+
+class TestKefMediaPlayer(unittest.TestCase):
+    """Unit tests for kef component."""
+
+    # pylint: disable=invalid-name
+    def setUp(self):
+        """Set up things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+        self.config = {
+            DOMAIN: {
+                CONF_PLATFORM: 'kef',
+                CONF_HOST: '192.0.1.1',
+            }
+        }
+        self.turn_on_config = {
+            DOMAIN: {
+                CONF_PLATFORM: 'kef',
+                CONF_HOST: '192.0.1.1',
+                CONF_NAME: 'speaker',
+                CONF_TURN_ON_SERVICE: "homeassistant.turn_on",
+                CONF_TURN_ON_DATA: "{}"
+            }
+        }
+        self.devices = list()
+
+    # pylint: disable=invalid-name
+    def tearDown(self):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    @mock.patch('pykef.KefSpeaker', side_effect=KefSpeakerMock)
+    def test_setup(self, kef_speaker):
+        """Test a single device setup."""
+        setup_component(self.hass, DOMAIN, self.config)
+        devices = list(self.hass.data[DOMAIN].entities)
```

Comment:
Please assert things about the state of the entity, not the entity itself. Get the state of the entity from the core state machine.

https://developers.home-assistant.io/docs/en/dev_101_hass.html